pkg/controllermanager/controller/controllerregistration/extensionclusterrole/reconciler.go

```diff
@@ -0,0 +1,124 @@
+}
+
+// Reconcile reconciles ControllerRegistrations.
+func (r *Reconciler) Reconcile(ctx context.Context, request reconcile.Request) (reconcile.Result, error) {
+	log := logf.FromContext(ctx)
+
+	ctx, cancel := controllerutils.GetMainReconciliationContext(ctx, controllerutils.DefaultReconciliationTimeout)
+	defer cancel()
+
+	clusterRole := &metav1.PartialObjectMetadata{}
+	clusterRole.SetGroupVersionKind(rbacv1.SchemeGroupVersion.WithKind("ClusterRole"))
+	if err := r.Client.Get(ctx, request.NamespacedName, clusterRole); err != nil {
+		if apierrors.IsNotFound(err) {
+			log.V(1).Info("Object is gone, stop reconciling")
+			return reconcile.Result{}, nil
+		}
+		return reconcile.Result{}, fmt.Errorf("error retrieving object from store: %w", err)
+	}
+
+	if clusterRole.DeletionTimestamp != nil {
+		return reconcile.Result{}, nil
+	}
+
+	subjects, err := r.computeSubjects(ctx, clusterRole)
+	if err != nil {
+		return reconcile.Result{}, fmt.Errorf("failed computing subjects for ClusterRoleBinding: %w", err)
+	}
+
+	clusterRoleBinding := &rbacv1.ClusterRoleBinding{ObjectMeta: metav1.ObjectMeta{Name: clusterRole.Name}}
+	_, err = controllerutils.GetAndCreateOrMergePatch(ctx, r.Client, clusterRoleBinding, func() error {
+		ownerReference := metav1.NewControllerRef(clusterRole, rbacv1.SchemeGroupVersion.WithKind("ClusterRole"))
+		ownerReference.BlockOwnerDeletion = pointer.Bool(false)
+		clusterRoleBinding.OwnerReferences = []metav1.OwnerReference{*ownerReference}
+
+		clusterRoleBinding.RoleRef = rbacv1.RoleRef{
+			APIGroup: rbacv1.SchemeGroupVersion.Group,
+			Kind:     "ClusterRole",
+			Name:     clusterRole.Name,
+		}
+		clusterRoleBinding.Subjects = subjects
+		return nil
+	})
+	return reconcile.Result{}, err
+}
+
+func (r *Reconciler) computeSubjects(ctx context.Context, clusterRole *metav1.PartialObjectMetadata) ([]rbacv1.Subject, error) {
+	labelSelector, err := clusterRoleServiceAccountLabelSelectorToSelector(clusterRole.Annotations)
+	if err != nil {
+		return nil, fmt.Errorf("failed parsing label selector: %w", err)
+	}
+
+	seedNamespaceList := &metav1.PartialObjectMetadataList{}
+	seedNamespaceList.SetGroupVersionKind(corev1.SchemeGroupVersion.WithKind("NamespaceList"))
+	if err := r.Client.List(ctx, seedNamespaceList, client.MatchingLabels{v1beta1constants.GardenRole: v1beta1constants.GardenRoleSeed}); err != nil {
+		return nil, fmt.Errorf("failed listing seed namespaces: %w", err)
+	}
+
+	// ensure stable order of subjects
+	kubernetesutils.ByName().Sort(seedNamespaceList)
+
+	var subjects []rbacv1.Subject
+
+	for _, namespace := range seedNamespaceList.Items {
+		serviceAccountList := &metav1.PartialObjectMetadataList{}
+		serviceAccountList.SetGroupVersionKind(corev1.SchemeGroupVersion.WithKind("ServiceAccountList"))
+		if err := r.Client.List(ctx, serviceAccountList, client.InNamespace(namespace.Name)); err != nil {
+			return nil, fmt.Errorf("failed listing ServiceAccounts in namespace %s: %w", namespace.Name, err)
```

Comment:
Shouldn't we use the label selector from the `ClusterRole` instead of iterating over the service account further below?